lib/src/components/Popover/Popover.tsx

```diff
@@ -0,0 +1,185 @@
+import { assignInlineVars } from "@vanilla-extract/dynamic";
+import {
+  forwardRef,
+  ForwardRefRenderFunction,
+  KeyboardEvent,
+  MutableRefObject,
+  useEffect,
+  useRef,
+  useState,
+} from "react";
+import { useClickOutside } from "../../hooks/useClickOutside";
+import { Flex } from "../Flex";
+import {
+  contentRecipe,
+  popArrowOffset,
+  popArrowSize,
+  popOffset,
+  popoverContainerStyles,
+  popRadius,
+  popWidth,
+} from "./popover.css";
+import { PopoverType } from "./popover.types";
+
+const PopoverComponent: ForwardRefRenderFunction<
+  HTMLDivElement,
+  PopoverType
+> = (
+  {
+    children,
+    position = "bottom",
+    content,
+    offset = "4px",
+    borderRadius = "4px",
+    width = "fit-content",
+    withArrow = false,
+    arrowSize = "7px",
+    isOpened,
+    onChange = () => {},
+    className,
+    style,
+    zIndex = "1",
+    trapFocus = false,
+  },
+  ref
+) => {
+  const [isOpen, setIsOpen] = useState<boolean | null>(null);
+  const [targetWidth, setTargetWidth] = useState("");
+  const popRef = useRef() as MutableRefObject<HTMLDivElement>;
+  const contentRef = useRef() as MutableRefObject<HTMLDivElement>;
+  const targetRef = useRef() as MutableRefObject<HTMLDivElement>;
+
+  useEffect(() => {
+    isOpened !== undefined && setIsOpen(isOpened);
+  }, [isOpened]);
+
+  useEffect(() => {
+    isOpen !== null && onChange(isOpen!);
+  }, [isOpen]);
+
+  useEffect(() => {
+    if (width === "target") {
+      setTargetWidth(getTargetWidth());
+    } else {
+      setTargetWidth(width);
+    }
+  }, [width, targetRef.current]);
+
+  useClickOutside(
+    popRef,
+    () => {
+      isOpened === undefined && setIsOpen(false);
+    },
+    isOpen!
+  );
+
+  const internalClickHandler = () => {
+    isOpened === undefined && setIsOpen(!isOpen);
+  };
+  const internalKeyDownHandler = (event: KeyboardEvent<HTMLDivElement>) => {
+    switch (event.code) {
+      case "Space":
+      case "Enter":
+        event.preventDefault();
+        internalClickHandler();
+        break;
+      case "Escape":
+        event.preventDefault();
+        isOpened === undefined && setIsOpen(false);
+        break;
+      default:
+        break;
+    }
+  };
+
+  const internalContentKeyDownHandler = (e: KeyboardEvent<HTMLDivElement>) => {
+    const focusableElements =
+      'button, [href], input, select, textarea, [tabindex]:not([tabindex="-1"])';
```

Comment:
pls save this in a global const so that we have access to this everywhere

Reply:
sure